### PR TITLE
Merge transactions with same hash when writing to event db

### LINKED ISSUE #2348: Duplicate transactions added to transaction table
Sometimes transactions copies of transactions are inserted into the [transaction table](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/dbs/event/transaction.go#L12).

### PR DESCRIPTION
**Pull request by Sriep**
## Fixes
Merge transactions with same hash when writing to event db
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/dbs/event/transaction_test.go [hunk 1]
```diff
@@ -13,33 +13,60 @@ import (
...
+		},
 	}
-	eventDb, err := NewEventDb(access, config.DbSettings{})
-	require.NoError(t, err)
-	defer eventDb.Close()
-	err = eventDb.AutoMigrate()
+	events, err := mergeEvents(round, "", transactionsEvents)
 	require.NoError(t, err)
+	require.Len(t, events, 1)
+	require.Len(t, events[0].Data, 3, "the five events should have been merged into three")
```
Comment (Kishan-Dhakan) [inline code/go/0chain.net/smartcontract/dbs/event/transaction_test.go:63]: Do we also expect the latest one to be the correct info? We can assert "two" has "Fee" 7 for example.

Comment (Sriep) [reply code/go/0chain.net/smartcontract/dbs/event/transaction_test.go:63]: In practice, it does not matter, as any duplicate transactions will always be identical.

**Review by dabasov — APPROVED** [review]
lgtm

**Review by peterlimg — APPROVED** [review]
LGTM

**Pull request merged**